demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -535,62 +576,55 @@ static bool readCredentials( SSL_CTX * pSslContext,
             if( pRootCa == NULL )
             {
                 LogError( ( "Failed to parse root CA." ) );
-                status = false;
+            }
+            else
+            {
+                sslStatus = 1;
+                LogDebug( ( "Successfully parsed root CA." ) );
             }
         }
 
-        if( status )
+        if( sslStatus == 1 )
         {
             /* Add the root CA to certificate store. */
-            if( X509_STORE_add_cert( SSL_CTX_get_cert_store( pSslContext ),
-                                     pRootCa ) != 1 )
+            sslStatus = X509_STORE_add_cert( SSL_CTX_get_cert_store( pSslContext ),
+                                             pRootCa );
+
+            if( sslStatus != 1 )
             {
                 LogError( ( "Failed to add root CA to certificate store." ) );
-                status = false;
-            }
-            else
-            {
-                LogInfo( ( "Successfully imported root CA." ) );
             }
         }
     }
 
-    if( ( status ) &&
+    if( ( sslStatus ) &&
```

Comment:
Not sure if SSL functions above only return 0 or 1, or if it's possible for them to return another value. Either way, I think it's better to explicitly compare to 1 here